zk/stages/stage_sequence_execute_data_stream.go

```diff
@@ -78,29 +83,38 @@ func (sbc *SequencerBatchStreamWriter) writeBlockDetailsToDatastream(verifiedBun
 	return checkedVerifierBundles, nil
 }
 
-func finalizeLastBatchInDatastreamIfNotFinalized(batchContext *BatchContext, batchState *BatchState, thisBlock uint64) error {
+func checkIfLastBatchIsSealed(batchContext *BatchContext) (bool, error) {
+	isLastEntryBatchEnd, err := batchContext.cfg.datastreamServer.IsLastEntryBatchEnd()
+	if err != nil {
+		return false, err
+	}
+
+	return isLastEntryBatchEnd, nil
+}
+
+func finalizeLastBatchInDatastreamIfNotFinalized(batchContext *BatchContext, batchState *BatchState, thisBlock uint64) (bool, error) {
```

Comment:
This function is longer used. Is this expected?

Reply:
Yup yup, the same code is basically replicated in the runBatchEnd function so as we're about to do that immediately after we detect we're in a bad state it made sense to cut it out

Reply:
The "finalizeLastBatchInDatastreamIfNotFinalized" is used when converting a RPC node to Sequencer node. The idea is to finalize the previous batch in the DS. So If we no longer need it, then its logic must be handled by something else.